From 552638c78e375b6bf23643994c18fd3abede242c Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 20 Jan 2022 04:31:09 +0200
Subject: [PATCH] [SW-69190] gaudi3: add mirror regs api

Add read and write apis for mirror registers. The write apis are
enabled only for projects which support mirror register configuration
(currently gaudi3). This is done using HL_MIRROR_CFG_BASE macro.

As all memory access functions are defined in common arcboot.h, these
apis are defined there to maintain uniformity, while the compile time
check for HL_MIRROR_CFG_BASE is to maintain compatibility.

Change-Id: I2c52c44661e535f8e8daf4e6e4c3bfc5bf767e9e
---

diff --git a/include/arcboot.h b/include/arcboot.h
index d980b12..4be5777 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -194,4 +194,17 @@
 	base_writel(val, HL_SCRATCHPAD_SRAM_BASE + offset);
 }
 
+#ifdef HL_MIRROR_CFG_BASE
+static inline uint32_t mirror_readl(uint32_t offset)
+{
+	uint32_t reg = base_readl(HL_MIRROR_CFG_BASE + offset);
+
+	return reg;
+}
+
+static inline void mirror_writel(uint32_t val, uint32_t offset)
+{
+	base_writel(val, HL_MIRROR_CFG_BASE + offset);
+}
+#endif /* HL_MIRROR_CFG_BASE */
 #endif /* _ARCBOOT_H_ */
